Add extra dict possible for send to Chromecast (#16268)
home-assistant/home-assistant.github.io: source/_integrations/media_player.markdown

| `entity_id`            |      yes | Target a specific media player. To target all media players, use `all`.                                                                                                                       |
| `media_content_id`     |       no | A media identifier. The format of this is integration dependent. For example, you can provide URLs to Sonos and Cast but only a playlist ID to iTunes.                   |
| `media_content_type`   |       no | A media type. Must be one of `music`, `tvshow`, `video`, `episode`, `channel` or `playlist`. For example, to play music you would set `media_content_type` to `music`. |

> | `extra`                |      yes | Extra dictionary data eg. title, thumb send to pychromecast. Possible values can be found below.

Reviewer: Small tweaks and removed the naming of the underlying library. It doesn't matter in this context.

```suggestion
| `extra`                |      yes | Extra dictionary data to send, e.g., title, thumbnail. Possible values can be found below.
```

> 
> #####Extra dictionary data

Reviewer: ```suggestion
##### Extra dictionary data

```

> Info taken from: https://github.com/home-assistant-libs/pychromecast/blob/master/pychromecast/controllers/media.py

Reviewer: I don't think this adds value for the end-user, as this is not developer documentation. Maybe just remove this.

| `title`            |      str | title of the media. |
| `thumb`            |      str | thumbnail image URL. |
| `current_time`            |      float | Seconds since beginning of content. If the content is live content, and position is not specifed, the stream will start at the live position. |
| `autoplay`            |bool | whether the media will automatically play. |
| `stream_type`            |str | describes the type of media artifact as one of the following: "NONE", "BUFFERED", "LIVE". |
| `subtitles`            |str | URL of subtitle file to be shown on chromecast. |
| `subtitles_lang`            |str | language for subtitles. |
| `subtitles_mime`            |str | mimetype of subtitles. |
| `subtitle_id`            |int | id of subtitle to be loaded. |
| `enqueue`            |bool | if True, enqueue the media instead of play it. |
| `media_info`            |dict | additional MediaInformation attributes not explicitly listed. |
| `metadata`            |dict | additional MediaInformation attributes not explicitly listed. |

> | `media_info`            |dict | media metadata object, one of the following: GenericMediaMetadata, MovieMediaMetadata, TvShowMediaMetadata, MusicTrackMediaMetadata, PhotoMediaMetadata. |

Reviewer: Some comments here:

- Sentences should start with a capital letter.
- `dict` is not a term we use in end-user documentation. Use `map` instead.
- `str` is not a term we use in end-user documentation. Use `string` instead.
- Code fence possible values with backticks ``` ` ```

For this section, you could consider using a configuration block instead of a table.

Author: I used configuration blocked, but couldn't figure out what type lint would pass. Is yaml as code, ok? or python?

Reviewer: What didn't work with it?

It is now non-standard, which we should avoid.

Author: sorry, I don't know why I was convinced that you meant to switch from table to code block.
anyway I switched to %configuration% block, unfortunately it won't build. any idea what am I doing wrong?

Docs:
https://developers.google.com/cast/docs/reference/messages#MediaData

> https://developers.google.com/cast/docs/reference/web_receiver/cast.framework.messages.MediaInformation

Reviewer: Don't use bare URL's.

Example of calling media_player service with title and image set:
```yaml

> media_content_type: music

Reviewer: Can we make this example a bit more complex? Add a bit more context. E.g., adding the service call to it.

Also, make sure to add a new line before starting a code fenced block.

| `extra`                |      yes | Extra dictionary data to send, e.g., title, thumbnail. Possible values can be found below.

##### Extra dictionary data

```json
title: string - Title of the media.
thumb: string - Thumbnail image URL.
current_time: float - Seconds since beginning of content. If the content is
    live content, and position is not specifed, the stream will start at the
    live position
autoplay: bool - Whether the media will automatically play.
stream_type: string - Describes the type of media artifact as one of the
    following: "NONE", "BUFFERED", "LIVE".
subtitles: str - URL of subtitle file to be shown on chromecast.
subtitles_lang: string - Language for subtitles.
subtitles_mime: string - Mimetype of subtitles.
subtitle_id: int - Id of subtitle to be loaded.
enqueue: bool - If True, enqueue the media instead of play it.
media_info: map - Additional MediaInformation attributes not explicitly listed.
metadata: map - Media metadata object, one of the following:
    GenericMediaMetadata, MovieMediaMetadata, TvShowMediaMetadata,
    MusicTrackMediaMetadata, PhotoMediaMetadata.
```

> Documentaion:
> [Google Dev Documentaion MediaData](https://developers.google.com/cast/docs/reference/messages#MediaData)
> [Google Dev Documentaion MediaInformation](https://developers.google.com/cast/docs/reference/web_receiver/cast.framework.messages.MediaInformation)

Reviewer: ```suggestion
Documentation:

- [Google Dev Documentaion MediaData](https://developers.google.com/cast/docs/reference/messages#MediaData)
- [Google Dev Documentaion MediaInformation](https://developers.google.com/cast/docs/reference/web_receiver/cast.framework.messages.MediaInformation)
```

| `entity_id`            |      yes | Target a specific media player. To target all media players, use `all`.                                                                                                                       |
| `media_content_id`     |       no | A media identifier. The format of this is integration dependent. For example, you can provide URLs to Sonos and Cast but only a playlist ID to iTunes.                   |
| `media_content_type`   |       no | A media type. Must be one of `music`, `tvshow`, `video`, `episode`, `channel` or `playlist`. For example, to play music you would set `media_content_type` to `music`. |
| `extra`                |      yes | Extra dictionary data to send, e.g., title, thumbnail. Possible values can be found below.

##### Extra dictionary data

{% configuration %}
title:
  type: string
  description: Title of the media.
  required: false
thumb:
  type: string
  description: Thumbnail image URL.
  required: false
current_time:
  type: float

>   description: Seconds since beginning of content. If the content is live content, and position is not specifed, the stream will start at the live position

Reviewer: ```suggestion
  description: Seconds since the beginning of the content. If the content is live content, and the position is not specified, the stream will start at the live position.
```

  required: false
autoplay:
  type: bool
  description: Whether the media will automatically play.
  required: false
stream_type:
  type: string

>   description: Describes the type of media artifact as one of the following: "NONE", "BUFFERED", "LIVE".

Reviewer: ```suggestion
  description: "Describes the type of media artifact as one of the following: `NONE`, `BUFFERED`, `LIVE`."
```

  required: false
subtitles:
  type: string
  description: URL of subtitle file to be shown on chromecast.
  required: false
subtitles_lang:
  type: string
  description: Language for subtitles.
  required: false
subtitles_mime:

>   type :string

Reviewer: ```suggestion
  type: string
```

  description: Mimetype of subtitles.
  required: false
subtitle_id:

>   type: int

Reviewer: ```suggestion
  type: integer
```

>   description: Id of subtitle to be loaded.

Reviewer: ```suggestion
  description: ID of subtitle to be loaded.
```

  required: false
enqueue:

>   type: bool

Reviewer: ```suggestion
  type: boolean
```

  description: If True, enqueue the media instead of play it.
  required: false
media_info:
  type: map
  description: Additional MediaInformation attributes not explicitly listed.
  required: false
metadata:
  type: map

>   description: Media metadata object, one of the following: GenericMediaMetadata, MovieMediaMetadata, TvShowMediaMetadata, MusicTrackMediaMetadata, PhotoMediaMetadata.

Reviewer: ```suggestion
  description: "Media metadata object, one of the following: `GenericMediaMetadata`, `MovieMediaMetadata`, `TvShowMediaMetadata`, `MusicTrackMediaMetadata`, `PhotoMediaMetadata`."
```

However, since this is a map, it should be listed as key instead.
